sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorIndexSpec.java

```diff
@@ -66,6 +74,127 @@ public CosmosVectorIndexSpec setType(String type) {
         checkNotNull(type, "cosmosVectorIndexType cannot be null");
```

Comment:
Types vs untyped for supported values. 
In .NET we are using typed(enum) to represent.

Untyped do give better flexibility in-case we want to delegate all validation to the service. 
But any consistency of set of properties also needs some understanding of the contract also.

Reply:
Added a check for this. Updating this to typed will be a breaking change now.